### PR TITLE
Implement NSDictionary+NSFileManagerExtensions

### LINKED ISSUE #2429: Move NSFileManagerExtensions to its own extension header and implementation
And mark as not-in-plan items that are not supportable on windows.

/* NSFileManager Extensions */
- (NSDate*)fileCreationDate;
- (BOOL)fileExtensionHidden STUB_METHOD;
- (NSNumber*)fileGroupOwnerAccountID STUB_METHOD;
- (NSString*)fileGroupOwnerAccountName STUB_METHOD;
- (OSType)fileHFSCreatorCode STUB_METHOD;
- (OSType)fileHFSTypeCode STUB_METHOD;
- (BOOL)fileIsAppendOnly STUB_METHOD;
- (BOOL)fileIsImmutable STUB_METHOD;
- (NSDate*)fileModificationDate;
- (NSNumber*)fileOwnerAccountID STUB_METHOD;
- (NSString*)fileOwnerAccountName STUB_METHOD;
- (NSUInteger)filePosixPermissions STUB_METHOD;
- (unsigned long long)fileSize;
- (NSUInteger)fileSystemFileNumber STUB_METHOD;
- (NSInteger)fileSystemNumber STUB_METHOD;
- (NSString*)fileType;

### PR DESCRIPTION
**Pull request by aballway**
Moves previously implemented NSFileManager extension methods and newly implemented methods to separate header and class category. This change does not change the NSFileManager methods which are generally used to populate NSDictionaries with the values these methods retrieve.

Fixes #2429

### CODE REVIEW

File: Frameworks/Foundation/NSDictionary+NSFileManagerExtensions.mm [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ @Status Interoperable
+*/
+- (NSUInteger)filePosixPermissions {
+    return [[self objectForKey:NSFilePosixPermissions] unsignedIntegerValue];
+}
+
+/**
+ @Status Interoperable
+*/
+- (unsigned long long int)fileSize {
```
Comment (DHowett-MSFT) [inline Frameworks/Foundation/NSDictionary+NSFileManagerExtensions.mm]: nit: explicit int not required

File: Frameworks/Foundation/NSDictionary+NSFileManagerExtensions.mm [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    unsigned long long int ret = [[self objectForKey:NSFileSize] unsignedLongLongValue];
```
Comment (DHowett-MSFT) [inline Frameworks/Foundation/NSDictionary+NSFileManagerExtensions.mm]: no need for the temporary?

File: include/Foundation/NSDictionary+NSFileManagerExtensions.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#include <Foundation/NSDictionary.h>
```
Comment (DHowett-MSFT) [inline include/Foundation/NSDictionary+NSFileManagerExtensions.h]: you'll likely want to either include headers for or forward-declare for `NSNumber`, `NSDate`, `NSString`, `OSType`, `BOOL`, `NSInteger` and `NSUInteger`

File: include/Foundation/NSDictionary+NSFileManagerExtensions.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+- (BOOL)fileIsImmutable;
+- (NSDate*)fileModificationDate;
+- (NSNumber*)fileOwnerAccountID;
+- (NSString*)fileOwnerAccountName;
+- (NSUInteger)filePosixPermissions;
+- (unsigned long long int)fileSize;
+- (NSUInteger)fileSystemFileNumber;
+- (NSInteger)fileSystemNumber;
+- (NSString*)fileType;
+@end
```
Comment (DHowett-MSFT) [inline include/Foundation/NSDictionary+NSFileManagerExtensions.h]: I'd be willing to bet these are all in `NSFileManager.h` on OS X: somebody including just NSFileManager would probably want to use them.

File: build/Headers/Headers.Shared/Public.Shared.vcxitems [hunk 5]
```diff
@@ -39,7 +39,6 @@
     <Text Include="$(MSBuildThisFileDirectory)..\..\..\include\CoreVideo\*.h" />
     <Text Include="$(MSBuildThisFileDirectory)..\..\..\include\EventKit\*.h" />
     <Text Include="$(MSBuildThisFileDirectory)..\..\..\include\EventKitUI\*.h" />
-    <Text Include="$(MSBuildThisFileDirectory)..\..\..\include\Foundation\*.h" />
```
Comment (ms-jihua) [inline build/Headers/Headers.Shared/Public.Shared.vcxitems]: ? context?

File: build/build.sln [hunk 6]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.26228.4
```
Comment (ms-jihua) [inline build/build.sln]: ?

File: build/Headers/Headers.Shared/Public.Shared.vcxitems (hunk 5 above)
Comment (aballway) [inline build/Headers/Headers.Shared/Public.Shared.vcxitems]: I've no clue when/why this changed.  Will revert these files w/ no changes

File: build/build.sln (hunk 6 above)
Comment (aballway) [inline build/build.sln]: Apparently Visual Studio went wild 😆

File: Frameworks/Foundation/NSDictionary+NSFileManagerExtensions.mm [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+ @Status Interoperable
+*/
+- (NSString*)fileGroupOwnerAccountName {
+    return [self objectForKey:NSFileGroupOwnerAccountName];
+}
+
+/**
+ @Status Interoperable
+*/
+- (OSType)fileHFSCreatorCode {
```
Comment (rajsesh-msft) [inline Frameworks/Foundation/NSDictionary+NSFileManagerExtensions.mm]: I don't see how many of these methods can be interoperable  Caveat them please.

**Pull request merged**